### PR TITLE
Order pdf renew

### PR DESCRIPTION
**Pull request by lqdung-lockon**

### CODE REVIEW

File: Event/CommonEvent.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     * @param string $part
+     *
+     * @return string
+     */
+    public function renderPosition($html, $part)
+    {
+        // For old and new ec-cube version
+        // Search group
+        // Group 1
+        $search = '/(<li\s+id="dropmenu"[\s\S]*)'; // Points to start the search.
```
Comment (lammn) [inline Event/CommonEvent.php]: don't use inline comment

File: Event/CommonEvent.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     * Render position.
+     *
+     * @param string $html
+     * @param string $part
+     *
+     * @return string
+     */
+    public function renderPosition($html, $part)
+    {
+        // For old and new ec-cube version
```
Comment (lammn) [inline Event/CommonEvent.php]: group comment

File: Event/CommonEvent.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        // Search group
+        // Group 1
+        $search = '/(<li\s+id="dropmenu"[\s\S]*)'; // Points to start the search.
+        // Group 2
+        $search .= '(<ul\s+class="dropdown\-menu"[\s\S]*)'; // start drop down section.
+        // Group 3
+        $search .= '(<\/li>[\n\s]*<\/ul>)'; // The end of the dropdown section.
```
Comment (lammn) [inline Event/CommonEvent.php]: don't use inline comment

File: Event/CommonEvent.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        // Group 4
+        $search .= '([\s\S]*<form\s+id="dropdown\-form")/'; // Points to end the search.
```
Comment (lammn) [inline Event/CommonEvent.php]: don't use inline comment

File: Event/OrderPdfLegacy.php [hunk 5]
```diff
@@ -1,96 +1,96 @@
...
             // その他メニューに追加するバージョン
-            $form  = $crawler->filter('#dropmenu .dropdown-menu')->last()->html();
-            $newForm = $form . $parts;
+            $form = $crawler->filter('#dropmenu .dropdown-menu')->last()->html();
+            $newForm = $form.$parts;
 
-            $html = str_replace($form, $newForm , $html);
-         } catch (\InvalidArgumentException $e) {
+            $html = str_replace($form, $newForm, $html);
+        } catch (\InvalidArgumentException $e) {
```
Comment (lammn) [inline Event/OrderPdfLegacy.php:80]: log error when catch

File: OrderPdfEvent.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        $eventLegacy = $this->app['eccube.plugin.order_pdf.event.order_pdf_legacy'];
+        $eventLegacy->onRenderAdminOrderPdfBefore($event);
+    }
+
+    /**
+     * Check to support new hookpoint.
+     *
+     * @return bool v3.0.9以降のフックポイントに対応しているか？
+     */
+    private function supportNewHookPoint()
```
Comment (lammn) [inline OrderPdfEvent.php:75]: Move this function to utils.

File: Resource/doctrine/migration/Version201611111300.php [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+     */
+    public function down(Schema $schema)
+    {
+        if (Version::isSupportGetInstanceFunction()) {
+            $app = Application::getInstance();
+            $meta = $this->getMetadata($app['orm.em']);
+            $tool = new SchemaTool($app['orm.em']);
+            $schemaFromMetadata = $tool->getSchemaFromMetadata($meta);
+            // テーブル削除
+            foreach ($schemaFromMetadata->getTables() as $table) {
```
Comment (lammn) [inline Resource/doctrine/migration/Version201611111300.php:64]: add code for remove sequence in postgresql

Comment (lqdung-lockon) [reply Resource/doctrine/migration/Version201611111300.php:64]: Have not sequence table.

File: Resource/doctrine/migration/Version201611111300.php [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+     */
+    protected function createOrderPdf(Schema $schema)
+    {
+        if ($schema->hasTable(self::TABLE)) {
+            return true;
+        }
+
+        $app = Application::getInstance();
+        $em = $app['orm.em'];
+        $classes = array(
```
Comment (ryo-endo) [inline Resource/doctrine/migration/Version201611111300.php:91]: can you use `$this->getMetadata` function ?

File: ServiceProvider/OrderPdfServiceProvider.php [hunk 9]
```diff
@@ -1,90 +1,127 @@
...
+        // 強制SSL
+        if ($app['config']['force_ssl'] == Constant::ENABLED) {
+            $admin->requireHttps();
+        }
+        // 帳票の作成
+        $admin->match('/plugin/order-pdf', '\\Plugin\\OrderPdf\\Controller\\OrderPdfController::index')
+            ->bind('plugin_admin_order_pdf');
+
+        // PDFファイルダウンロード
+        $admin->match('/plugin/order-pdf/download', '\\Plugin\\OrderPdf\\Controller\\OrderPdfController::download')
```
Comment (ryo-endo) [inline ServiceProvider/OrderPdfServiceProvider.php]: $admin->post ?

File: event.yml [hunk 10]
```diff
@@ -1,16 +1,14 @@
...
-eccube.event.render.admin_order_page.before:
-    - [onRenderAdminOrderPdfBefore, NORMAL]
-
+#
+# Event fock
+#
+
+# New hookpoint for version >= 3.0.9
+Admin/Order/index.twig:
+    - [onAdminOrderRender, NORMAL]
```
Comment (ryo-endo) [inline event.yml]: onAdminOrderRender -> onAdminOrderIndexRender

File: PluginManager.php [hunk 11]
```diff
@@ -1,48 +1,145 @@
...
+    /**
+     * Enable.
+     *
+     * @param array       $config
+     * @param Application $app
+     */
+    public function enable($config, $app)
+    {
+        // Backup logo.
+        $this->backupLogo($app['config']);
```
Comment (ryo-endo) [inline PluginManager.php]: necessary ? (because, install function done this)

File: PluginManager.php [hunk 12]
```diff
@@ -1,48 +1,145 @@
...
+     * Update.
+     *
+     * @param array       $config
+     * @param Application $app
+     */
+    public function update($config, $app)
+    {
+        $arrConfig = $app['config'];
+        // Rollback to old logo
+        $this->rollBackLogo($arrConfig);
```
Comment (ryo-endo) [inline PluginManager.php]: update. I would like to keep the logo file. (maybe, customized to the user's shop logo)

File: PluginManager.php [hunk 13]
```diff
@@ -1,48 +1,145 @@
...
+
+    /**
+     * Roll back to old logo.
+     *
+     * @param array $config
+     */
+    private function rollBackLogo($config)
+    {
+        $file = new Filesystem();
+        $sourcePath = $config['template_realdir'].'/'.$this->logoName;
```
Comment (ryo-endo) [inline PluginManager.php]: its not write in best practice...
put in plugin code dir.
i.e.) $config['template_realdir'].'/OrderPdf/'.$this->logoName;

**Review by ryo-endo — CHANGES_REQUESTED** [review]

File: Resource/doctrine/migration/Version201611111300.php (hunk 8 above)
Comment (lqdung-lockon) [reply Resource/doctrine/migration/Version201611111300.php:91]: I do like the Maker plugin! Is it wrong?

File: ServiceProvider/OrderPdfServiceProvider.php (hunk 9 above)
Comment (lqdung-lockon) [reply ServiceProvider/OrderPdfServiceProvider.php]: Yes, I will fix it.

File: event.yml (hunk 10 above)
Comment (lqdung-lockon) [reply event.yml]: I think the default is "index" so do not need.

File: PluginManager.php (hunk 13 above)
Comment (lqdung-lockon) [reply PluginManager.php]: Yes, I will fix it.

File: PluginManager.php (hunk 11 above)
Comment (ryo-endo) [reply PluginManager.php]:
```
public function install($config, $app)
{
    // Backup logo.
    $this->backupLogo($app['config']);
}
```

```
public function enable($config, $app)
{
    // Backup logo.
    $this->backupLogo($app['config']);
    
    $this->migrationSchema($app, __DIR__.'/Resource/doctrine/migration', $config['code']);
}
```

`$this->backupLogo($app['config']);` is duplicated.
i think, its unnecessary in `function enable`.

File: PluginManager.php (hunk 12 above)
Comment (ryo-endo) [reply PluginManager.php]: Update：
if (logo.png is exist) {
    don't overwrite logo file. (keep user's logo file)
} else {
   //not exist
   copy the logo.png
}

File: ServiceProvider/OrderPdfServiceProvider.php [hunk 14]
```diff
@@ -1,90 +1,127 @@
...
+        // ============================================================
+        // メッセージ登録
+        // ============================================================
+        $file = __DIR__.'/../Resource/locale/message.'.$app['locale'].'.yml';
+        if (file_exists($file)) {
+            $app['translator']->addResource('yaml', $file, $app['locale']);
+        }
+
+        // Config
+        $app['config'] = $app->share($app->extend('config', function ($config) {
```
Comment (kiy0taka) [inline ServiceProvider/OrderPdfServiceProvider.php]: Use config.yml to define constants.
See http://ec-cube.github.io/plugin_bp_config#section-2

**Review by kiy0taka — CHANGES_REQUESTED** [review]

**Review by ryo-endo — CHANGES_REQUESTED** [review]
### [bug]
カレンダーが2重に表示されてしまいます。
他の画面と同じ表示・動きになるのが期待値です。
image

File: event.yml (hunk 10 above)
Comment (ryo-endo) [reply event.yml]: "index" not means "default page" (like index.html).
it means ... "index search" "index card" "index page" ...index.

i'd like to rename to `onAdminOrderIndexRender` to clarify.

File: Resource/doctrine/migration/Version201611111300.php (hunk 8 above)
Comment (ryo-endo) [reply Resource/doctrine/migration/Version201611111300.php:91]: i read code again.
sorry, its my mistake. no problem.

File: PluginManager.php (hunk 11 above)
Comment (ryo-endo) [reply PluginManager.php]: fix https://github.com/eccubevn/order-pdf-plugin/pull/3

File: PluginManager.php (hunk 12 above)
Comment (ryo-endo) [reply PluginManager.php]: https://github.com/eccubevn/order-pdf-plugin/pull/3

**Review by ryo-endo — CHANGES_REQUESTED** [review]
### [bug ]
複数受注を選択して、帳票出力ができない。
(463e356 から発生している)

File: Resource/template/admin/order_pdf.twig [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+                $("input[name='admin_order_pdf[issue_date]']").val('{{ "now"|date("Y-m-d") }}');
+                $("input[name='admin_order_pdf[message1]']").val('{{ form.vars.value.message1 }}');
+                $("input[name='admin_order_pdf[message2]']").val('{{ form.vars.value.message2 }}');
+                $("input[name='admin_order_pdf[message3]']").val('{{ form.vars.value.message3 }}');
+                $("input[name='admin_order_pdf[note1]']").val('{{ form.vars.value.note1 }}');
+                $("input[name='admin_order_pdf[note2]']").val('{{ form.vars.value.note2 }}');
+                $("input[name='admin_order_pdf[note3]']").val('{{ form.vars.value.note3 }}');
+            });
+
+            $("#back").click(function (event) {
```
Comment (ryo-endo) [inline Resource/template/admin/order_pdf.twig]: ### [bug]

戻るの実装が簡易的すぎて、検索結果に戻れないパターンがあります。
他のページ（たとえば受注編集画面）と合わせてください。
難しい実装ではないので、修正お願いします。

**Review by ryo-endo — CHANGES_REQUESTED** [review]

**Review by ryo-endo — APPROVED** [review]

**Pull request closed without merging**